changeset:   4557:4aa31b9207f2
branch:      SDL-1.2
parent:      4554:4deaba8b1b42
user:        Sam Lantinga <slouken@libsdl.org>
date:        Tue Jul 20 00:01:26 2010 -0700
summary:     Finally fixed bug 894 without breaking bug 716.  Yay! :)

diff -r 4deaba8b1b42 -r 4aa31b9207f2 src/video/x11/SDL_x11events.c
--- a/src/video/x11/SDL_x11events.c	Sun Jul 18 11:37:12 2010 -0700
+++ b/src/video/x11/SDL_x11events.c	Tue Jul 20 00:01:26 2010 -0700
@@ -441,8 +441,10 @@
 if ( xevent.xcrossing.mode == NotifyUngrab )
 printf("Mode: NotifyUngrab\n");
 #endif
-		if ( xevent.xcrossing.detail != NotifyInferior ) {
-			if ( this->input_grab == SDL_GRAB_OFF ) {
+		if ( (xevent.xcrossing.mode != NotifyGrab) &&
+		     (xevent.xcrossing.mode != NotifyUngrab) &&
+		     (xevent.xcrossing.detail != NotifyInferior) ) {
+               		if ( this->input_grab == SDL_GRAB_OFF ) {
 				posted = SDL_PrivateAppActive(0, SDL_APPMOUSEFOCUS);
 			} else {
 				posted = SDL_PrivateMouseMotion(0, 0,

